.github/workflows/main.yml

```
@@ -0,0 +1,123 @@
+name: main
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: read
+
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [
+          "py-27",
+          "py-33",
+          "py-34",
+          "py-35",
+          "py-36",
+          "py-37",
+          "py-38",
+          "py-39",
+          "pypy-36",
+          "pypy-27"
+        ]
+
+        include:
+          - name: "py-27"
+            python: "2.7"
+            os: ubuntu-latest
+          - name: "py-33"
+            python: "3.3"
+            os: ubuntu-latest
+            container: true
+          - name: "py-34"
+            python: "3.4"
+            os: ubuntu-latest
+            container: true
+          - name: "py-35"
+            python: "3.5"
+            os: ubuntu-latest
+          - name: "py-36"
+            python: "3.6"
+            os: ubuntu-latest
+          - name: "py-37"
+            python: "3.7"
+            os: ubuntu-latest
+          - name: "py-38"
+            python: "3.8"
+            os: ubuntu-latest
+          - name: "py-39"
+            python: "3.9"
+            os: ubuntu-latest
+          - name: "pypy-36"
+            python: "pypy-3.6-v7.3.2"
+            os: ubuntu-latest
+            container: true
+          - name: "pypy-27"
+            python: "pypy-2.7.13-v7.3.2"
+            os: ubuntu-latest
+            container: true
+
+    steps:
+    - uses: actions/checkout@v2
```

[v4](https://github.com/actions/checkout/releases/tag/v4) is the latest version of `actions/checkout` currently.
```suggestion
    - uses: actions/checkout@v4
```